### PR TITLE
[DependencyInjection] Fix a wrong error when using a factory

### PR DESCRIPTION
**Pull request by Simperfit**
…d a call

| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #30885    
| License       | MIT
| Doc PR        |

This a work in progress, it needs tests, firstly I wanted to ask the author to test and tell if it's ok.

### CODE REVIEW

**Review by jderusse — APPROVED** [review]
Reproduced on my app environment.
Work for me

File: src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php [hunk 1]
```diff
@@ -158,8 +158,15 @@ private function autowireCalls(\ReflectionClass $reflectionClass, bool $isRoot):
...
-            }
+                $definition = new Definition($reflectionClass->name);
+                try {
+                    $reflectionMethod = $this->getReflectionMethod($definition, $method);
+                } catch (RuntimeException $e) {
+                    if ($definition->getFactory()) {
+                        continue;
+                    }
+                    throw $e;
+                }            }
```
Comment (dmaicher) [inline src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: weird formatting :smile:

Comment (Simperfit) [reply src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php]: oops

**Review by tgalopin — APPROVED** [review]

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Should probably be applied on 3.4 or 4.2

**Comment by fabpot** [comment]
Thank you @Simperfit.

**Pull request merged**